core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -134,6 +134,10 @@ public boolean handle(int code) {
             // we've already logged this.
         }
 
+        if (Constants.LINUX) {
+            JNANatives.maxNumberOfProcesses();
```

Comment:
@rmuir Yes, I agree. This whole situation is silly since POSIX does *not* define `RLIMIT_NPROC`, BSD does, Linux *did* derive `RLIMIT_NPROC` from BSD, but changed the meaning when it did so. :frowning: 

I pushed a rename in 8fb30e11424c6ffcdc010cd31e41d8c7ee7250c7.